src/Dialog/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,192 @@
+import React, {useCallback} from 'react'
+import ReactDOM from 'react-dom'
+import styled from 'styled-components'
+import {Box, ThemeProvider, useTheme} from '..'
+import {FocusKeys} from '../behaviors/focusZone'
+import {get} from '../constants'
+import {Dialog, DialogProps, DialogHeaderProps, DialogButtonProps} from '../Dialog/Dialog'
+import {useFocusZone} from '../hooks/useFocusZone'
+import {ThemeProviderProps} from '../ThemeProvider'
+
+/**
+ * Props to customize the ConfirmationDialog.
+ */
+export interface ConfirmationDialogProps {
+  /**
+   * Required. This callback is invoked when a gesture to close the dialog
+   * is performed. The first argument indicates the gesture.
+   */
+  onClose: (gesture: 'confirm' | 'cancel' | 'close-button' | 'cancel' | 'escape') => void
+
+  /**
+   * Required. The title of the ConfirmationDialog. This is usually a brief
+   * question.
+   */
+  title: React.ReactNode
+
+  /**
+   * The text to use for the cancel button. Default: "Cancel".
+   */
+  cancelButtonContent?: React.ReactNode
+
+  /**
+   * The text to use for the confirm button. Default: "OK".
+   */
+  confirmButtonContent?: React.ReactNode
+
+  /**
+   * The type of button to use for the confirm button. Default: Button.
+   */
+  confirmButtonType?: 'normal' | 'primary' | 'danger'
+}
+
+const StyledConfirmationHeader = styled.header`
+  padding: ${get('space.2')};
+  display: flex;
+  flex-direction: row;
+`
+const StyledTitle = styled(Box)`
+  font-size: ${get('fontSizes.3')};
+  font-weight: 700;
+  padding: 6px ${get('space.2')};
+  flex-grow: 1;
+`
+const ConfirmationHeader: React.FC<DialogHeaderProps> = ({title, onClose, dialogLabelId}) => {
+  const onCloseClick = useCallback(() => {
+    onClose('close-button')
+  }, [onClose])
+  return (
+    <StyledConfirmationHeader>
+      <StyledTitle id={dialogLabelId}>{title}</StyledTitle>
+      <Dialog.CloseButton onClose={onCloseClick} />
+    </StyledConfirmationHeader>
+  )
+}
+const StyledConfirmationBody = styled(Box)`
+  font-size: ${get('fontSizes.1')};
+  padding: 0 ${get('space.3')} 22px ${get('space.3')};
```

Comment:
```suggestion
  padding: 0 ${get('space.3')} ${get('space.3')} ${get('space.3')};
```

The Figma mockup had `22px` as a total calculation considering this would use an `OverlayHeader` to generate the Title + description (8px padding + original 32px total size minus the text line-height). Since this is a one-off, let's keep it at `space.3`. ✌️